tests/test_client_functional.py

```diff
@@ -3022,6 +3022,30 @@ async def handler(request):
             await resp.read()
 
 
+async def test_timeout_with_full_buffer(aiohttp_client: Any) -> None:
+    async def handler(request):
+        """Server response that never ends and always has more data available."""
+        resp = web.StreamResponse()
+        await resp.prepare(request)
+        while True:
+            await resp.write(b"1" * 1000)
+            await asyncio.sleep(0.01)
+
+    async def request(client):
+        timeout = aiohttp.ClientTimeout(total=0.5)
+        with pytest.raises(asyncio.TimeoutError):
+            async with await client.get("/", timeout=timeout) as resp:
+                async for data in resp.content.iter_chunked(1):
+                    await asyncio.sleep(0.01)
+
+    app = web.Application()
+    app.add_routes([web.get("/", handler)])
+
+    client = await aiohttp_client(app)
+    # wait_for() timeout should not be reached.
```

Comment:
If it shouldn't be reached, use `raise RuntimeError` or `pytest.fail()` instead..

Reply:
Sorry, I meant the timeout shouldn't be reached. I used wait_for() just to ensure that a failing test doesn't hang forever.